server/app/views/admin/programs/ProgramBlockPredicateConfigureView.java

```diff
@@ -0,0 +1,621 @@
+        {
+          throw new RuntimeException(
+              String.format(
+                  "Unrecognized predicate format: %s", existingPredicate.computePredicateFormat()));
+        }
+    }
+  }
+
+  private DivTag renderQuestionHeaders(
+      ImmutableList<QuestionDefinition> questionDefinitions,
+      Optional<PredicateDefinition> maybeExistingPredicate) {
+    DivTag container = div().withClasses("flex", "py-4");
+    int columnNumber = 1;
+
+    if (maybeExistingPredicate.isPresent()) {
+      ImmutableList<LeafOperationExpressionNode> leafNodes =
+          getExistingAndNodes(maybeExistingPredicate.get()).stream()
+              .findFirst()
+              .get()
+              .getAndNode()
+              .children()
+              .stream()
+              .map(PredicateExpressionNode::getLeafNode)
+              .sorted(
+                  (LeafOperationExpressionNode loenA, LeafOperationExpressionNode loenB) ->
+                      (int) (loenA.questionId() - loenB.questionId()))
+              .collect(ImmutableList.toImmutableList());
+
+      for (LeafOperationExpressionNode leafNode : leafNodes) {
+        long questionId = leafNode.questionId();
+        QuestionDefinition qd =
+            questionDefinitions.stream()
+                .filter(questionDefinition -> questionDefinition.getId() == questionId)
+                .findFirst()
+                .get();
+
+        container.with(
+            div(
+                    div(qd.getQuestionText().getDefault())
+                        .withClasses(BaseStyles.INPUT, "text-gray-500", "mb-2", "truncate"),
+                    createScalarDropdown(qd, Optional.of(leafNode.scalar())),
+                    createOperatorDropdown(
+                        qd, Optional.of(leafNode.scalar()), Optional.of(leafNode.operator())))
+                .withClasses(COLUMN_WIDTH, columnNumber++ != 1 ? "ml-16" : null));
+      }
+    } else {
+      for (var qd : questionDefinitions) {
+        container.with(
+            div(
+                    div(qd.getQuestionText().getDefault())
+                        .withClasses(BaseStyles.INPUT, "text-gray-500", "mb-2", "truncate"),
+                    createScalarDropdown(qd, /* maybeScalar */ Optional.empty()),
+                    createOperatorDropdown(
+                        qd,
+                        /* maybeSelectedScalar */ Optional.empty(),
+                        /* maybeOperator */ Optional.empty()))
+                .withClasses(COLUMN_WIDTH, columnNumber++ != 1 ? "ml-16" : null));
+      }
+    }
+
+    return container.with(div().withClasses("w-28"));
+  }
+
+  private DivTag renderValueRow(
+      ImmutableList<QuestionDefinition> questionDefinitions,
+      int groupId,
+      Optional<AndNode> maybeAndNode) {
+    DivTag row = div().withClasses("flex", "mb-6", "predicate-config-value-row");
+    DivTag andText = div("and").withClasses("object-center", "w-16", "p-4", "leading-10");
+    int columnNumber = 1;
```

Comment:
Make a unique one in each if/else branch

Reply:
done